lib/new_relic/agent/tracer.rb

```diff
@@ -409,15 +409,18 @@ def clear_state
 
         alias_method :tl_clear, :clear_state
 
-        def thread_block_with_current_transaction(*args, &block)
+        def thread_block_with_current_transaction(*args, segment_name:, &block)
           current_txn = ::Thread.current[:newrelic_tracer_state].current_transaction if ::Thread.current[:newrelic_tracer_state] && ::Thread.current[:newrelic_tracer_state].is_execution_traced?
           proc do
             begin
               if current_txn
                 NewRelic::Agent::Tracer.state.current_transaction = current_txn
-                segment = NewRelic::Agent::Tracer.start_segment(name: "Ruby/Thread/#{::Thread.current.object_id}")
+                segment_name += "/Thread#{::Thread.current.object_id}/Fiber#{::Fiber.current.object_id}" if NewRelic::Agent.config[:'thread_ids_enabled']
```

Comment:
We refactored this method so that it could be used by both concurrent ruby and threads. (also fibers in future!). 
We also removed the thread id from the segment name to prevent thread related MGIs from occurring. I really wanted to option to turn on thread ids because it's helped me out with figuring issues out in the past, so we added the new `thread_ids_enabled` so that we can turn it on whenever we think it would help out.